jbmc/src/java_bytecode/load_method_by_regex.cpp

```diff
@@ -28,43 +27,50 @@ load_method_by_regext::load_method_by_regext(const std::string &pattern)
   if(!has_prefix(pattern, "java::"))
     modified_pattern = "java::" + modified_pattern;
 
-  regex = std::regex{modified_pattern};
-}
-
-/// Get the methods that match the regex
-/// \param symbol_table: The symbol table to look through
-/// \return: The method identifiers that should be loaded
-std::vector<irep_idt>
-load_method_by_regext::extra_methods(const symbol_tablet &symbol_table) const
-{
-  std::vector<irep_idt> matched_methods;
-  for(const auto &symbol : symbol_table.symbols)
-  {
-    if(symbol.second.is_function())
-    {
-      const std::string symbol_name = id2string(symbol.first);
-      if(std::regex_match(symbol_name, regex))
-      {
-        matched_methods.push_back(symbol_name);
-      }
-    }
-  }
-  return matched_methods;
+  return std::regex{modified_pattern};
 }
 
 /// Identify if a parameter includes a part that will match a descriptor. That
 /// is, does it have a colon separtor.
 /// \param pattern: The user provided pattern
 /// \return True if no descriptor is found (that is, the only : relates to the
 /// java:: prefix.
-bool load_method_by_regext::does_pattern_miss_descriptor(
-  const std::string &pattern)
+bool does_pattern_miss_descriptor(const std::string &pattern)
 {
   const size_t descriptor_index = pattern.rfind(':');
   if(descriptor_index == std::string::npos)
     return true;
 
   static const std::string java_prefix = "java::";
   return descriptor_index == java_prefix.length() - 1 &&
-         has_prefix(pattern, "java::");
+         has_prefix(pattern, java_prefix);
+}
+
+/// Create a lambda that returns the symbols that the given pattern should be
+/// loaded.If the pattern doesn't include a colon for matching the descriptor,
+/// append a `:\(.*\).*` to the regex. Note this will mean all overloaded
+/// methods will be marked as extra entry points for CI lazy loading.
+/// If the pattern doesn't include the java:: prefix, prefix that
+/// \param pattern: The user provided pattern
+/// \return The lambda to execute.
+std::function<std::vector<irep_idt>(const symbol_tablet &symbol_table)>
+build_load_method_by_regex(const std::string &pattern)
+{
+  std::regex regex = build_regex_from_pattern(pattern);
+
+  return [=](const symbol_tablet &symbol_table) {
+    std::vector<irep_idt> matched_methods;
+    for(const auto &symbol : symbol_table.symbols)
+    {
+      if(symbol.second.is_function())
+      {
+        const std::string symbol_name = id2string(symbol.first);
+        if(std::regex_match(symbol_name, regex))
+        {
+          matched_methods.push_back(symbol_name);
+        }
```

Comment:
if(symbol.second.is_function()
     	  && std::regex_match(id2string(symbol.first), regex))
      {
          matched_methods.push_back(symbol.first);
      }

Reply:
Sure if you like, not convinced it is any more readable (And I'm not removing braces :upside_down_face: )